core/capabilities/webapi/outgoing_connector_handler.go

```diff
@@ -60,22 +68,22 @@ func NewOutgoingConnectorHandler(gc connector.GatewayConnector, config ServiceCo
 	if !validMethod(method) {
 		return nil, fmt.Errorf("invalid outgoing connector handler method: %s", method)
 	}
-
-	return &OutgoingConnectorHandler{
+	och := &OutgoingConnectorHandler{
 		gc:                  gc,
 		method:              method,
 		responses:           newResponses(),
 		outgoingRateLimiter: outgoingRateLimiter,
 		incomingRateLimiter: incomingRateLimiter,
 		lggr:                lgger,
 		selectorOpts:        opts,
-	}, nil
+	}
+	och.toRetryableFunc = och.toRetryable
+
+	return och, nil
 }
 
 // HandleSingleNodeRequest sends a request to first available gateway node and blocks until response is received
-// TODO: handle retries
 func (c *OutgoingConnectorHandler) HandleSingleNodeRequest(ctx context.Context, messageID string, req capabilities.Request) (*api.Message, error) {
```

Comment:
"Make the change easy, then make the easy change" -- we're getting to a point where this function is insufficiently abstracted out which is making your change hard to do.

I would:
- isolate the core logic of making the request into a synchronous function called something like sendRequest -- including your rate limit logic, gateway selection logic, send + select logic (this is basically everything there is today, with the exception of the timeout)
- layer on top of that any changes relating to retries

For testability you might consider implementing a helper that implements an interface you can mock out.